partner_risk_insurance/wizard/invoice_risk_insurance_wizard.py

```diff
@@ -13,31 +13,30 @@ class InvoiceRiskInsuranceWizard(models.TransientModel):
         string="From", required=True, default=fields.Date.context_today
     )
     date_to = fields.Date(string="To", required=True, default=fields.Date.context_today)
+    only_unpaid = fields.Boolean(
+        help="If check, includes only invoices in state not paid"
+    )
+
+    def get_moves_domain(self):
+        domain = [
+            ("state", "=", "posted"),
+            ("move_type", "in", ("out_invoice", "out_refund")),
+            ("date", ">=", self.date_from),
+            ("date", "<=", self.date_to),
+        ]
+        if self.only_unpaid:
+            domain.append(("payment_state", "=", "not_paid"))
+        return domain
 
     def action_print_report(self):
-        moves = self.env["account.move"].search(
-            [
-                ("state", "=", "posted"),
-                ("move_type", "in", ("out_invoice", "out_refund")),
-                ("payment_state", "=", "not_paid"),
-                ("date", ">=", self.date_from),
-                ("date", "<=", self.date_to),
-            ]
-        )
+        moves = self.env["account.move"].search(self.get_moves_domain())
         if not moves:
             raise ValidationError(
                 _(
                     "There aren't moves between this dates\n"
                     "Please, select different dates"
                 )
             )
-        moves = moves.sorted(
```

Comment:
Isn't this sort not needed anymore?

Reply:
Sorted in template because seems that the order is lost in the report_action call:
https://github.com/OCA/credit-control/blob/39119989c54e1198d17403372f032170604b1ce9/partner_risk_insurance/templates/invoice_risk_insurance_template.xml#L21-L22